src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -72,7 +73,7 @@ public ComplaintResponse createComplaintResponse(ComplaintResponse complaintResp
         }
         // Only tutors who are not the original assessor of the submission can reply to a complaint
         if (!authorizationCheckService.isAtLeastTeachingAssistantForExercise(originalComplaint.getResult().getParticipation().getExercise())
-                || originalComplaint.getResult().getAssessor().equals(reviewer)) {
+                || (originalComplaint.getResult().getAssessor().equals(reviewer) && originalComplaint.getComplaintType().equals(ComplaintType.COMPLAINT))) {
```

Comment:
Maybe I'm mistaken, but aren't there existing complaints that don't have a complaint type, as this attribute was introduced in this PR?
If I'm correct, this could cause a NullPointerException.